pkg/provider/gcp/secretmanager/auth.go

```diff
@@ -33,7 +33,6 @@ func NewTokenSource(ctx context.Context, auth esv1beta1.GCPSMAuth, projectID str
 	}
 	wi, err := newWorkloadIdentity(ctx, projectID)
 	if err != nil {
-		useMu.Unlock()
```

Comment:
removing this unlock was intentional? Wouldn't the code gets forever locked if `newWorkLoadIdentity` fails? 🙈